extensions/cache/deployment/src/main/java/io/quarkus/cache/deployment/CacheProcessor.java

```diff
@@ -232,17 +233,26 @@ private List<Throwable> validateKeyGeneratorsDefaultConstructor(CombinedIndexBui
 
     @BuildStep
     @Record(RUNTIME_INIT)
-    SyntheticBeanBuildItem configureCacheManagerSyntheticBean(CacheNamesBuildItem cacheNames,
-            CacheManagerRecorder cacheManagerRecorder, Optional<MetricsCapabilityBuildItem> metricsCapability) {
+    void cacheManagerInfos(BuildProducer<CacheManagerInfoBuildItem> producer,
+            Optional<MetricsCapabilityBuildItem> metricsCapability, CacheManagerRecorder recorder) {
+        producer.produce(new CacheManagerInfoBuildItem(recorder.noOpCacheManagerInfo()));
+        producer.produce(new CacheManagerInfoBuildItem(recorder.getCacheManagerInfoWithoutMetrics()));
```

Comment:
Is this needed when Micrometer is available?

Reply:
We could certainly remove it, but it makes no real difference as the supplier is only invoked when necessary